### ISSUE TITLE
Error installing `python setup.py egg_info`

### ISSUE THREAD

**Issue by Integralist** [jmdana/memprof#23]
Hello,

I've tried installing for Python `3.6.1` and `3.6.0` and both error.

```
pip install --upgrade memprof
```

## Error Installing

```
Command "python setup.py egg_info" failed with error code 1 in /private/var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/pip-build-ize31kqg/memprof/
```

## Stack Trace

```
$ pip install --upgrade memprof
Collecting memprof
  Using cached memprof-0.3.4.tar.gz
    Complete output from command python setup.py egg_info:
    Searching for Cython
    Reading https://pypi.python.org/simple/Cython/
    Downloading https://pypi.python.org/packages/10/32/21873ff231e069f860098b1602bb9e3ae2806d2f73ba661b5d806f200243/Cython-0.27.1.tar.gz#md5=b93750b34382b025d678bea819b7491b
    Best match: Cython 0.27.1
    Processing Cython-0.27.1.tar.gz
    Writing /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/setup.cfg
    Running Cython-0.27.1/setup.py -q bdist_egg --dist-dir /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/egg-dist-tmp-w8zvr71p
    Compiling /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Plex/Scanners.py because it changed.
    Compiling /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Plex/Actions.py because it changed.
    Compiling /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Compiler/Pythran.py because it changed.
    Compiling /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Compiler/Lexicon.py because it changed.
    Compiling /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Compiler/Scanning.py because it changed.
    Compiling /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Compiler/Parsing.py because it changed.
    Compiling /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Compiler/Visitor.py because it changed.
    Compiling /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Compiler/FlowControl.py because it changed.
    Compiling /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Compiler/Code.py because it changed.
    Compiling /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Runtime/refnanny.pyx because it changed.
    Compiling /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Tempita/_tempita.py because it changed.
    [ 1/11] Cythonizing /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Compiler/Code.py
    [ 2/11] Cythonizing /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Compiler/FlowControl.py
    [ 3/11] Cythonizing /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Compiler/Lexicon.py
    [ 4/11] Cythonizing /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Compiler/Parsing.py
    [ 5/11] Cythonizing /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Compiler/Pythran.py
    [ 6/11] Cythonizing /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Compiler/Scanning.py
    [ 7/11] Cythonizing /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Compiler/Visitor.py
    [ 8/11] Cythonizing /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Plex/Actions.py
    [ 9/11] Cythonizing /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Plex/Scanners.py
    [10/11] Cythonizing /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Runtime/refnanny.pyx
    [11/11] Cythonizing /var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/easy_install-ahreh7zt/Cython-0.27.1/Cython/Tempita/_tempita.py
    Unable to find pgen, not compiling formal grammar.
    warning: no files found matching '2to3-fixers.txt'
    warning: no files found matching 'Doc/*'
    warning: no files found matching '*.pyx' under directory 'Cython/Debugger/Tests'
    warning: no files found matching '*.pxd' under directory 'Cython/Debugger/Tests'
    warning: no files found matching '*.pxd' under directory 'Cython/Utility'
    creating /Users/markmcdonnell/.pyenv/versions/3.6.0/lib/python3.6/site-packages/Cython-0.27.1-py3.6-macosx-10.12-x86_64.egg
    Extracting Cython-0.27.1-py3.6-macosx-10.12-x86_64.egg to /Users/markmcdonnell/.pyenv/versions/3.6.0/lib/python3.6/site-packages
    Adding Cython 0.27.1 to easy-install.pth file
    Installing cygdb script to /Users/markmcdonnell/.pyenv/versions/3.6.0/bin
    Installing cython script to /Users/markmcdonnell/.pyenv/versions/3.6.0/bin
    Installing cythonize script to /Users/markmcdonnell/.pyenv/versions/3.6.0/bin

    Installed /Users/markmcdonnell/.pyenv/versions/3.6.0/lib/python3.6/site-packages/Cython-0.27.1-py3.6-macosx-10.12-x86_64.egg
    Processing dependencies for Cython
    Finished processing dependencies for Cython
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/pip-build-ize31kqg/memprof/setup.py", line 40, in <module>
        class md_easy_install(easy_install):
    TypeError: module.__init__() takes at most 2 arguments (3 given)

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /private/var/folders/3m/3pnsckx15wgdv5cpynbtz4b40000gn/T/pip-build-ize31kqg/memprof/
```

**Comment by jmdana** [comment]
Does it work if you create the environment with Python 3.5 instead?

**Comment by Integralist** [comment]
I had other issues with my Python environment and so I scrapped the whole thing and restarted all environments I had installed.

This helped me with some other packages I was having trouble installing.

I now only have one Python version installed `3.6.3` (the exception being the macOS system python).

But now if I try to install this package it just hangs:

```
$ pip install memprof
Collecting memprof
  Using cached memprof-0.3.4.tar.gz
```
